sql/moz-fx-data-shared-prod/telemetry_derived/cohort_daily_statistics_v2/query.sql

```diff
@@ -0,0 +1,150 @@
+    partner_distributor_channel,
+    partner_id,
+    city,
+    country,
+    -- device_model, -- important for mobile, usually null, may not need
+    apple_model_id, -- from clients_first_seen_v2
+    distribution_id,
+    -- is_default_browser,  -- can find in clients_daily_v6
+    locale,
+    -- normalized_app_name,  -- strings that come from unified_metrics/filters out BrowserStack
+    app_name as normalized_app_name, -- from clients_first_seen_v2, as per Lucia's msg app_name = normalized_app_name
+    normalized_channel,
+    normalized_os,
+    normalized_os_version,
+    -- os_version_major,
+    -- os_version_minor,
+    COALESCE(
+      SAFE_CAST(NULLIF(SPLIT(normalized_os_version, ".")[SAFE_OFFSET(0)], "") AS INTEGER),
+      0
+    ) AS os_version_major,
+    COALESCE(
+      SAFE_CAST(NULLIF(SPLIT(normalized_os_version, ".")[SAFE_OFFSET(1)], "") AS INTEGER),
+      0
+    ) AS os_version_minor,
+  FROM
+    telemetry_derived.clients_first_seen_v2
+  WHERE
+    first_seen_date
+    BETWEEN DATE_SUB(@activity_date, INTERVAL 180 DAY)
+    AND DATE_SUB(@activity_date, INTERVAL 1 DAY)
+    -- (1) No need to get activity for the cohort created on activity_date - everyone will be retained
+    -- (2) Note this is a pretty big scan... Look here for problems
+),
+activity_cohort_match AS (
+  SELECT
+    cohorts_in_range.client_id AS client_id,
+    submission_date_activity.client_id AS active_client_id,
+    submission_date_activity.days_seen_bits as active_client_days_seen_bits,
+    mozfun.bits28.days_since_seen(submission_date_activity.days_seen_bits) as active_clients_days_since_seen,
+    cohorts_in_range.* EXCEPT (client_id)
+  FROM
+    cohorts_in_range
+  LEFT JOIN
+    submission_date_activity
+  USING
+    (client_id, activity_date)
+)
+SELECT
+  cohort_date,
+  activity_date,
+  -- activity_segment,
+  app_version,
+  attribution_campaign,
+  attribution_content,
+  attribution_experiment,
+  attribution_medium,
+  attribution_source,
+  attribution_ua,
+  -- attribution_variation,
+  partner_distribution_version,
+  partner_distributor,
+  partner_distributor_channel,
+  partner_id,
+  city,
+  country,
+  -- device_model,
+  apple_model_id, -- from clients_first_seen_v2
+  distribution_id,
+  -- is_default_browser,
+  locale,
+  normalized_app_name,
+  normalized_channel,
+  normalized_os,
+  normalized_os_version,
+  os_version_major,
+  os_version_minor,
+  COUNT(client_id) AS num_clients_in_cohort,
+  COUNTIF((active_client_id IS NOT NULL) AND (active_clients_days_since_seen = 0)) AS num_clients_active_on_day,
+  COUNTIF((active_client_id IS NOT NULL) AND
+    (COALESCE(
+    BIT_COUNT(mozfun.bits28.from_string('0000000000000000000001111111') & active_client_days_seen_bits) > 0,
+    FALSE
+    )) ) AS num_clients_active_atleastonce_in_last_7_days,
+  COUNTIF((active_client_id IS NOT NULL) AND
+    ( COALESCE(
+    BIT_COUNT(mozfun.bits28.from_string('0111111111111111111111111111') & active_client_days_seen_bits) > 0,
```

Comment:
This is a slightly tricky bit. I think we want this column to count all clients active in the past 28 days, but the `mozfun.bits28.from_string('0111111111111111111111111111') &` part will ignore the first day in that 28 day period (so its only looking at 27 days actually). 

The original intention of this code from `clients_first_seen_28_days_later` is to calculate so-called first month repeat users, which ignores the client's first day, and only counts them if they show up for an _additional day in their first 28 days_. 

I think we would want to capture both first month repeat users AND 28-day retention in this table. Probably the best way to do that would be to change this column to remove the bitmask, i.e.  

```sql
COUNTIF((active_client_id IS NOT NULL) AND
    ( COALESCE(
    BIT_COUNT(active_client_days_seen_bits) > 0,
    FALSE) )) AS num_clients_active_atleastonce_in_last_28_days
```
Then, add a new column, like

```sql
COUNTIF((active_client_id IS NOT NULL) AND
    DATE_DIFF(activity_date, cohort_date, DAY) <= 27 AND
    ( COALESCE(
    BIT_COUNT(mozfun.bits28.from_string('0111111111111111111111111111') & active_client_days_seen_bits) > 0,
    FALSE) )) AS num_clients_repeat_first_month_users
```
This new condition should only be TRUE in cases where clients are active at least 2 days anytime in their first 28 days.

Reply:
this makes sense to me. Will update. Any objections @ANich ?